chain/db/db.go

```diff
@@ -349,3 +358,45 @@ func (b *BandDB) HandleMessage(txHash []byte, msg sdk.Msg, events map[string]str
 
 	return jsonMap, nil
 }
+
+func (b *BandDB) GetInvolvedAccounts(tx auth.StdTx) []sdk.AccAddress {
+	involvedAccounts := make([]sdk.AccAddress, 0)
+	for _, msg := range tx.GetMsgs() {
+		switch msg := msg.(type) {
+		case zoracle.MsgRequestData:
+			involvedAccounts = append(involvedAccounts, msg.Sender)
+		case zoracle.MsgReportData:
+			involvedAccounts = append(involvedAccounts, msg.Reporter)
+		case bank.MsgSend:
+			involvedAccounts = append(involvedAccounts, msg.FromAddress, msg.ToAddress)
+		case bank.MsgMultiSend:
+			for _, input := range msg.Inputs {
+				involvedAccounts = append(involvedAccounts, input.Address)
+			}
+			for _, output := range msg.Outputs {
+				involvedAccounts = append(involvedAccounts, output.Address)
+			}
+		case staking.MsgCreateValidator:
+			involvedAccounts = append(involvedAccounts, msg.DelegatorAddress)
+		case staking.MsgDelegate:
+			involvedAccounts = append(involvedAccounts, msg.DelegatorAddress)
+		case staking.MsgBeginRedelegate:
+			involvedAccounts = append(involvedAccounts, msg.DelegatorAddress)
+		case staking.MsgUndelegate:
+			involvedAccounts = append(involvedAccounts, msg.DelegatorAddress)
+		case dist.MsgWithdrawDelegatorReward:
+			involvedAccounts = append(involvedAccounts, msg.DelegatorAddress)
+		case dist.MsgWithdrawValidatorCommission:
+			involvedAccounts = append(involvedAccounts, sdk.AccAddress(msg.ValidatorAddress))
+		case gov.MsgDeposit:
+			involvedAccounts = append(involvedAccounts, msg.Depositor)
+		case gov.MsgSubmitProposal:
+			involvedAccounts = append(involvedAccounts, msg.Proposer)
+		default:
```

Comment:
Now that we handle a lot of messages. Let's make sure we have case for all messages, and panic for unknown messages, so we fail fast.